File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = [];
    
    count = 1;
    a = 100 - progresses[0];
    a = a / speeds[0];
    a = math.ceil(a)
    
    for i in range(1,len(progresses)) :

        b = 100 - progresses[i];
        b = b / speeds[i];
        b = math.ceil(b)

        if a >= b :
            count += 1;
        elif a < b :
            answer.append(count);
            count = 1;
            a = b;
    answer.append(count);
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let i = 1;
    while (progresses.length > 0) {
        
        if (progresses[0] + i * speeds[0] >= 100) {
            progresses.shift();
            speeds.shift();
            answer.push(i);
        }
        else {
            i++;  
        }
    }
    
    let count = 1;
    let day = answer[0];
    let stack = [];

    for(let i=1;i<answer.length;i++) {
        if(answer[i] === day) {
            count++;
        }    
        else {
            stack.push(count);
            count = 1;
            day = answer[i];
        }
    }
    stack.push(count);
    return stack;
}